pkg/vault/client_test.go

```diff
@@ -111,6 +112,82 @@ func TestGetKV1Secret(t *testing.T) {
 	})
 }
 
+func TestWriteKV2Secret(t *testing.T) {
+	t.Parallel()
+	const secretAPIPath = "secret/data/test"
+
+	t.Run("Test write new KV2 secret", func(t *testing.T) {
+		vaultMock := &mocks.VaultMock{}
+		client := Client{vaultMock, &Config{}}
+		setupMockKvV2(vaultMock)
+		vaultMock.On("Read", secretAPIPath).Return(nil, nil)
+		secret := map[string]interface{}{"data": map[string]interface{}{"key": "value"}}
+		vaultMock.On("Write", secretAPIPath, secret).Return(nil, nil)
+		err := client.WriteKvSecret("secret/test", map[string]string{"key": "value"})
+		assert.NoError(t, err)
```

Comment:
What does your test actually do assert? I think assert.NoError() is not enough to validate the behaviour of the write function. I think you should ensure that Write() was actually called.

Reply:
the same is true for all tests below.